File: BE_QuanLyTro/src/main/java/ut/edu/be_quanlytro/Dto/Request/DepositUpdateRequest.java
package ut.edu.be_quanlytro.Dto.Request;

import lombok.Getter;
import lombok.Setter;
import ut.edu.be_quanlytro.Entity.Enum.DepositStatus;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@Setter
public class DepositUpdateRequest {
    private String phone;
    private String tenantFullName;
    private BigDecimal depositAmount;
    private LocalDate expectedMoveInDate;
    private String note;
    private DepositStatus status; // Cho phép chủ trọ đổi trạng thái (VD: Hủy cọc)
}

File: Flutter_QuanLyTro/lib/data/models/request/deposit_update_request.dart
class DepositUpdateRequest {
  final String phone;
  final String tenantFullName;
  final double depositAmount;
  final String expectedMoveInDate;
  final String? note;
  final String status;

  DepositUpdateRequest({
    required this.phone,
    required this.tenantFullName,
    required this.depositAmount,
    required this.expectedMoveInDate,
    this.note,
    required this.status,
  });

  Map<String, dynamic> toJson() {
    return {
      'phone': phone,
      'tenantFullName': tenantFullName,
      'depositAmount': depositAmount,
      'expectedMoveInDate': expectedMoveInDate,
      'note': note,
      'status': status,
    };
  }
}